flake.nix

```diff
@@ -90,104 +84,76 @@
             };
           })
       ];
-
-      projectFor = system:
+    in
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        ./pre-commit.nix
+        ./hercules-ci.nix
+      ];
+      flake = {
+        haskellModules = perSystem haskellModules;
+      };
+      systems = defaultSystems;
+      perSystem = { system, config, pkgs, ... }:
         let
-          pkgs = nixpkgsFor system;
-          pkgs' = nixpkgsFor' system;
+          project =
+            pkgs.haskell-nix.cabalProject {
+              name = "plutip-core";
+              src = ./.;
+              inputMap = {
+                "https://input-output-hk.github.io/cardano-haskell-packages" = CHaP;
+              };
+              compiler-nix-name = "ghc8107";
+
+              shell = {
+                withHoogle = true;
+                exactDeps = true;
+
+                tools.haskell-language-server = "1.5.0.0"; # Newer versions failed to build
```

Comment:
TODO: Include std haskell tooling + whatever you need

```
              tools = {
                cabal = { };
                haskell-language-server = { };
                hlint = { };
                cabal-fmt = { };
                fourmolu = { };
              };
```

Reply:
I specifically removed these in this pr because now `pre-commit` provides those.
I guess `cabal` should be there though.

Reply:
Actually, pre-commit doesn't have the same version of these tools.

But more importantly, you need to add them so IDEs have access to them. And for that, sure you could take out the pre-commit ones, but rather I use the latest from haskell.nix

Reply:
Ideally we would have the tools aligned, so you dev environment and pre-commit have access to the same versions. For now, this works and doesn't complicate the flake which is important.